src/test/java/me/karakelley/http/utility/HtmlGeneratorTest.java

```diff
@@ -0,0 +1,32 @@
+package me.karakelley.http.utility;
+
+import me.karakelley.http.ContentGeneration.HtmlGenerator;
+import me.karakelley.http.FileSystem.FileFinderProxy;
+import me.karakelley.http.FileSystem.PublicDirectory;
+import me.karakelley.http.FileSystem.RealFileFinder;
+import me.karakelley.http.helpers.TempFilesHelper;
+import org.junit.jupiter.api.Test;
+
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class HtmlGeneratorTest {
+
+  @Test
+  void testCreateLink() {
+    HtmlGenerator htmlGenerator = new HtmlGenerator();
+    TempFilesHelper.withTempDirectory(directory -> {
+      Path pathOne = TempFilesHelper.createTempFile(directory);
+
+      assertTrue(htmlGenerator.displayDirectories(pathOne.toFile(), PublicDirectory.create("/", new FileFinderProxy(new RealFileFinder()))).contains("<p><a href="));
```

Comment:
For a test like this I'd probably prefer to find a way to assert *exactly* what I wanted to see as the return value.

```
assertEquals("<p><a href="/some/path/here">file-name.txt</a></p>")
```

I feel like every time you make a test dynamic like this you increase the risk of making the same bad assumption in both the assertion and the production code, which leads to bugs. It's not always bad, though. You just need to be aware of those risks.

I expect it'll be hard to test because this depends on an actual file being on the filesystem. At this level, it might be better to make it more of a "pure function" (something that takes a value and returns a value) rather than something that depends on global state (the fileystem in this case).

Reply:
I did fix this in the next PR for opening text files. I altered the `TempFilesHelper` to give me more concrete names so I can make better assertions (just like you suggested).